server/reload.go

```diff
@@ -115,21 +233,31 @@ func (s *Server) diffOptions(newOpts *Options) ([]option, error) {
 		}
 		switch strings.ToLower(field.Name) {
 		case "trace":
-			diffOpts = append(diffOpts, &traceOption{newValue.(bool)})
+			diffOpts = append(diffOpts, &traceOption{newValue: newValue.(bool)})
 		case "debug":
-			diffOpts = append(diffOpts, &debugOption{newValue.(bool)})
+			diffOpts = append(diffOpts, &debugOption{newValue: newValue.(bool)})
 		case "tlsconfig":
-			diffOpts = append(diffOpts, &tlsOption{newValue.(*tls.Config)})
+			diffOpts = append(diffOpts, &tlsOption{newValue: newValue.(*tls.Config)})
 		case "tlstimeout":
-			// TLSTimeout change is picked up when Options is swapped.
-			continue
+			diffOpts = append(diffOpts, &tlsTimeoutOption{newValue: newValue.(float64)})
+		case "username":
+			diffOpts = append(diffOpts, &usernameOption{})
+		case "password":
+			diffOpts = append(diffOpts, &passwordOption{})
+		case "authorization":
+			diffOpts = append(diffOpts, &authorizationOption{})
+		case "authtimeout":
+			diffOpts = append(diffOpts, &authTimeoutOption{newValue: newValue.(float64)})
+		case "users":
+			diffOpts = append(diffOpts, &usersOption{newValue: newValue.([]*User)})
 		case "port":
 			// check to see if newValue == 0 and continue if so.
 			if newValue == 0 {
 				// ignore RANDOM_PORT
 				continue
 			} else {
-				return nil, fmt.Errorf("Config reload not supported for %s", field.Name)
+				return nil, fmt.Errorf("Config reload not supported for %s: old=%v, new=%v",
+					field.Name, oldValue, newValue)
```

Comment:
Neither of these seem to cause reload to fail. Also changing the cluster port changes the entire ClusterOpts on the Options (which does cause a reload failure since we don't support ClusterOpts reload yet).